mindmeld/tokenizer.py

```diff
@@ -105,19 +106,26 @@ def _init_regex(self):
             "apostrophe": (" ", None),
         }
 
+        # Check if custom pattern is being used or MM defined
+        if self.config.get("allowed_patterns"):
+            self._custom = True
+
         # Create compiled regex expressions
         try:
             self.keep_special_compiled = re.compile(
                 "(%s)"
                 % (
-                    ")|(".join(self.config["allowed_patterns"]),
+                    ")|(".join(
+                        self.config["allowed_patterns"]
+                        or self.config["default_allowed_patterns"]
+                    ),
                 ),
-                re.UNICODE
+                re.UNICODE,
             )
         except sre_constants.error:
             logger.error(
                 "Custom regex compilation failed for the following patterns: %s",
-                ")|(".join(self.config["allowed_patterns"])
+                ")|(".join(self.config["allowed_patterns"]),
```

Comment:
Is it possible for some pathway to add characters to `exclude_from_norm` that would then make the default_allowed_patterns a bad regex? It may be better to not assume this error came from the custom pattern. Then you could do the string join once at the start/before the try block, and reuse it here.